nibabel/nifti1.py

```diff
@@ -1621,12 +1621,159 @@ def set_xyzt_units(self, xyz=None, t=None):
             xyz = 0
         if t is None:
             t = 0
-        xyz_code = self.structarr['xyzt_units'] % 8
-        t_code = self.structarr['xyzt_units'] - xyz_code
         xyz_code = unit_codes[xyz]
         t_code = unit_codes[t]
         self.structarr['xyzt_units'] = xyz_code + t_code
 
+    def get_zooms(self, units=None, raise_unknown=False):
+        ''' Get zooms (spacing between voxels along each axis) from header
+
+        NIfTI-1 headers may specify that zooms are encoded in units other than
+        mm and sec (see ``get_xyzt_units``).
+        Default behavior has been to return the raw zooms, and leave it to the
+        programmer to handle non-standard units.
+        However, most files indicate mm/sec units or have unspecified units,
+        and it is common practice to neglect specified units and assume all
+        files will be in mm/sec.
+
+        The default behavior for ``get_zooms`` will remain to return the raw
+        zooms until version 4.0, when it will change to return zooms in
+        normalized mm/sec units.
+        Because the default behavior will change, a warning will be given to
+        prompt programmers to specify whether they intend to retrieve raw
+        values, or values coerced into normalized units.
+
+        Parameters
+        ----------
+        units : {'norm', 'raw'}
+            Return zooms in normalized units of mm/sec for spatial/temporal or
+            as raw values stored in header.
+        raise_unkown : bool, optional
+            If normalized units are requested and the units are ambiguous, raise
+            a ``ValueError``
+
+        Returns
+        -------
+        zooms : tuple
+            tuple of header zoom values
+
+        '''
+        if units is None:
+            units = 'raw'
+            warnings.warn('Units not specified in `{}.get_zooms`. Returning '
+                          'raw zooms, but default will change to normalized.\n'
```

Comment:
might be worth announcing in what version this change is to happen (4.0)?

Also I am not sure if it would be best to keep default of get_zooms here (`raw`) be different from all other ones (`norm`)

Reply:
I'm okay with following whatever schedule is considered best. If I understand the overall philosophy, it's okay to break the API at 3.0; it's just that if we were planning to remove something entirely we'd need to schedule final removal for 4.0. I could be wrong though.

Also, the reason to keep this as `raw` for NIFTI is that anybody who has been correctly handling non-mm/s units will find they're suddenly double-correcting if we start correcting for them. I do agree that diverging defaults is dissatisfying, but it seemed least bad at the time. Open to other approaches.